### ISSUE TITLE
False positive when using generic protocol in overload signature

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5285]
From https://github.com/pandas-dev/pandas-stubs/issues/729:

```python
from typing import overload, Protocol, TypeVar, Any

AnyStr_contra = TypeVar("AnyStr_contra", str, bytes, contravariant=True)

class Buffer(Protocol):
    def __bytes__(self) -> bytes:
        ...

class BytesIO:
    def write(self, __b: Buffer) -> None:
        pass

class WriteBuffer(Protocol[AnyStr_contra]):
    def write(self, __b: AnyStr_contra) -> Any:
        ...

class NDFrame:
    @overload
    def to_csv(self, path_or_buf: WriteBuffer[bytes]) -> None:
        ...

    @overload
    def to_csv(self, path_or_buf: None = ...) -> str:
        ...

    def to_csv(self, path_or_buf: Any = None) -> Any:
        ...

def test_types_to_csv() -> None:
    df = NDFrame()
    df.to_csv(BytesIO())
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.315, which I just published. It will also be included in this week's insiders release of pylance.
